### PR TITLE
Set mail get mail updates

### PR DESCRIPTION
**Pull request by andrevka**
new types introdues following jmap specification ([https://jmap.io/spec.html](https://jmap.io/spec.html)):

1. IEmailSetProperties
2. IEmailSetResponse
3. ISetArguments
4. IGetEmailArguments

Add method for Email/set in the client

### CODE REVIEW

File: src/types.ts
Comment (alagane) [inline src/types.ts:19]: I wanted to use the type argument to tell `properties` is a list of keys of `Properties`, but I made an error defining it, it should have been `(keyof Properties)[]` instead of `Properties[]`, that is probably why you removed it?

Comment (andrevka) [reply src/types.ts:19]: I think so, i shall repair it

**Review by alagane — CHANGES_REQUESTED** [review]
Thanks for this pull request, this is really appreciated.
There are some changes I suggest, and it should be ready to merge.
I will rebase my refactor to apply it once this is merged (and probably the mailbox_set PR).

Don't hesitate to create issues if you think about some improvements for the project :wink:

**Review by alagane — APPROVED** [review]

**Comment by oubchid** [comment]
Thanks for this pull request
can you remove `package-lock.json` in your commit or add a `.gitignore` if not already added

**Comment by alagane** [comment]
> Thanks for this pull request
> can you remove `package-lock.json` in your commit or add a `.gitignore` if not already added

I think ignoring or deleting the package-lock.json is a bad idea.

You can also read this readme for Angular that can help understand how it can be useful (yarn.lock is like package-lock.json but for yarn instead of npm) https://github.com/angular/angular/blob/master/yarn.lock.readme.md

**Comment by oubchid** [comment]
I think it depends on project type. what works for project maintainers/ CI systems might not work when the project is used as a dependency. 
https://github.com/npm/npm/issues/20603

**Comment by alagane** [comment]
> I think it depends on project type. what works for project maintainers/ CI systems might not work when the project is used as a dependency.
> [npm/npm#20603](https://github.com/npm/npm/issues/20603)

This one really convinced me. Anyway we can remove it in a separate pull request since it was already present.

**Review by oubchid — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**
